src/protocol/prss.rs

```diff
@@ -153,49 +146,84 @@ impl Participant {
     /// # Panics
     /// When used incorrectly.  For instance, if you ask for an RNG and then ask
     /// for a PRSS using the same key.
-    pub fn prss(&self, key: impl AsRef<str>) -> &PrssSpace {
-        let p = self.inner.lock().unwrap().prss(key.as_ref());
+    pub fn indexed(&self, key: &UniqueStepId) -> &IndexedSharedRandomness {
+        let p = self.inner.lock().unwrap().indexed(key.as_ref());
         // Safety: This pointer refers to pinned memory that is allocated
         // and retained as long as this object exists.
         unsafe { p.as_ref() }
     }
+
+    /// Get a sequential shared randomness.
+    ///
+    /// # Panics
+    /// This can only be called once.  After that, calls to this function or `indexed` will panic.
+    pub fn sequential(
+        &self,
+        key: &UniqueStepId,
+    ) -> (SequentialSharedRandomness, SequentialSharedRandomness) {
+        self.inner.lock().unwrap().sequential(key.as_ref())
+    }
 }
 
-impl Debug for Participant {
+impl Debug for Endpoint {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.write_str("Participant")
     }
 }
 
-struct ParticipantInner {
+enum EndpointItem {
+    Indexed(Pin<Box<IndexedSharedRandomness>>),
+    Sequential,
+}
+
+struct EndpointInner {
     left: GeneratorFactory,
     right: GeneratorFactory,
     // TODO(mt): add a function to get an RNG instead of the indexed PRSS.
     // That should mark the entry as dead, so that any attempt to get the
     // indexed PRSS or another RNG will fail.
-    spaces: HashMap<String, Pin<Box<PrssSpace>>>,
+    items: HashMap<String, EndpointItem>,
 }
 
-impl ParticipantInner {
-    pub fn prss(&mut self, key: &str) -> NonNull<PrssSpace> {
+impl EndpointInner {
+    pub fn indexed(&mut self, key: &str) -> NonNull<IndexedSharedRandomness> {
         // The second arm of this statement would be fine, except that `HashMap::entry()`
         // only takes an owned value as an argument.
         // This makes the lookup perform an allocation, which is very much suboptimal.
-        let p = if let Some(value) = self.spaces.get(key) {
-            value
+        let item = if let Some(item) = self.items.get(key) {
+            item
         } else {
-            self.spaces.entry(key.to_owned()).or_insert_with_key(|k| {
-                Box::pin(PrssSpace {
+            self.items.entry(key.to_owned()).or_insert_with_key(|k| {
+                EndpointItem::Indexed(Box::pin(IndexedSharedRandomness {
```

Comment:
probably not for this PR, but I wonder if we should return a drop guard instead of reference to the value? I worry that we would keep all prss around for the duration of the query which may be billions if we have unique step per record. If the value returned is dropped, we can follow the backpointer to `Endpoint` and remove it from the map